examples/render/quad_render/main.rs

```diff
@@ -34,16 +33,16 @@ const QUAD: [Vertex; 6] = [
     Vertex { a_Pos: [ -0.5,-0.33 ], a_Uv: [0.0, 0.0] },
 ];
 
-gfx_descriptor_struct! {
-    SampleDesc {
+gfx_descriptors! {
+    desc {
         sampled_image: gfx::pso::SampledImage,
         sampler: gfx::pso::Sampler,
     }
 }
 
 gfx_graphics_pipeline! {
     pipe {
-        sample: gfx::pso::DescriptorSet<SampleDesc<B>>, // TODO remove <B>
+        desc: gfx::pso::DescriptorSet<desc::Set<B>>, // TODO: improve
```

Comment:
we still do need to get rid of `B` here since no other PSO component is backend dependent

Reply:
The macro could generate a `desc::Component`, how does that sound ?